### PR TITLE
Filesize optimitzations

### PR DESCRIPTION
**Pull request by jordic**
I upgraded it to latest cli and optimized some things for filesize.

### CODE REVIEW

File: Caddyfile [hunk 1]
```diff
@@ -0,0 +1,11 @@
+
```
Comment (hdjirdeh) [inline Caddyfile]: Not sure what this file is needed for, do we need this?

File: src/polyfills.ts [hunk 2]
```diff
@@ -1,19 +1,19 @@
...
+// import 'core-js/es6/string';
+// import 'core-js/es6/date';
+// import 'core-js/es6/array';
+// import 'core-js/es6/regexp';
+// import 'core-js/es6/map';
+// import 'core-js/es6/set';
+// import 'core-js/es6/reflect';
 
-import 'core-js/es7/reflect';
+// import 'core-js/es7/reflect';
```
Comment (hdjirdeh) [inline src/polyfills.ts]: I think we may need this specific import (reflect). Without this and serving the app I get this error:

`Uncaught reflect-metadata shim is required when using class decorators`

File: src/app/hackernews-api.service.ts [hunk 3]
```diff
@@ -1,28 +1,46 @@
 import { Injectable } from '@angular/core';
-import { Http } from '@angular/http';
+// import { Http } from '@angular/http';
 import { Observable } from 'rxjs/Observable';
 import 'rxjs/add/operator/map';
 
+declare var fetch;
+
+//wrap fetch in observable so we can keep it chill
+export function lazyFetch(url, options?){
```
Comment (hdjirdeh) [inline src/app/hackernews-api.service.ts]: This is awesome. Thank you so much! 

Only thing I noticed is that with this set up, the `onCompleted`doesn't fire in `stories.component.ts` when `fetchStories` is triggered. See [here](https://github.com/hdjirdeh/angular2-hn/blob/master/src/app/stories/stories.component.ts#L38-L39).

This means that the list numbering of items isn't correct, window doesn't scroll up when navigating as well as the checks on the `prev` and `next` (can click `prev` and `next` continuously with no checks in place). We just need to fix how we wrap  the completed event.

File: src/app/item/item.component.html [hunk 4]
```diff
@@ -1,62 +1,39 @@
...
-            <span *ngIf="item.comments_count > 1">comments</span>
-          </span>
-          <span *ngIf="item.comments_count === 0">discuss</span>
+          {{item.comments_count | comment }}
         </a>
       </span>
     </span>
   </div>
 </div>
-<div class="item-mobile">
```
Comment (hdjirdeh) [inline src/app/item/item.component.html]: Thank you thank you thank you, this cleans things up so nicely :)

File: src/manifest.webapp [hunk 5]
```diff
@@ -0,0 +1,29 @@
+{
```
Comment (hdjirdeh) [inline src/manifest.webapp]: Beauty, I've been meaning to add a manifest. Thanks!!!!!

File: Caddyfile (hunk 1 above)
Comment (jordic) [inline Caddyfile]: Sorry I committed it for error. Is the webserver config I'm using to test --aot builds

File: src/polyfills.ts (hunk 2 above)
Comment (jordic) [inline src/polyfills.ts]: Yes and no. With chrome you only need it on JIT. On --aot decorators are stripped. Also I will check the rest of polyfills needed for browser support

File: src/app/hackernews-api.service.ts (hunk 3 above)
Comment (jordic) [inline src/app/hackernews-api.service.ts]: I will fix it.. I replaced http module with this implementation I found on @robwormald repo before testing the case.

File: src/polyfills.ts (hunk 2 above)
Comment (hdjirdeh) [inline src/polyfills.ts]: Ah my bad, I just served the app without --aot and it failed. Should have checked serving with the decorator, but thank you appreciate it!

**Pull request merged**